### PR TITLE
add codegen test for equivalent slice implementations

### PR DESCRIPTION
**Pull request by reez12g**
addressing https://github.com/rust-lang/rust/issues/109328

### CODE REVIEW

File: tests/codegen/equivalent-slice-impls.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      [] => None,
+      [first, rest @ ..] => {
+          *input = rest;
+          Some(*first)
+      }
+  }
+}
+
+// CHECK-LABEL: @f2
+// CHECK: if
```
Comment (Mark-Simulacrum) [inline tests/codegen/equivalent-slice-impls.rs:18]: I would not have expected this to be enough to check that there are equal numbers of if instructions... don't need something that checks there's not a subsequent `if` too?

**Comment by Dylan-DPC** [comment]
@reez12g any updates on this?

**Comment by Dylan-DPC** [comment]
Closing this as inactive. Feel free to reöpen this pr or create a new pr if you get the time to work on this. Thanks

**Pull request closed without merging**